src/Analyzers/CSharp/Analyzers/UseImplicitOrExplicitType/CSharpTypeStyleDiagnosticAnalyzerBase.cs

```diff
@@ -42,9 +41,9 @@ public override bool OpenFileOnly(OptionSet options)
             var whereApparentOption = options.GetOption(CSharpCodeStyleOptions.VarWhenTypeIsApparent).Notification;
             var wherePossibleOption = options.GetOption(CSharpCodeStyleOptions.VarElsewhere).Notification;
 
-            return !(forIntrinsicTypesOption == NotificationOption.Warning || forIntrinsicTypesOption == NotificationOption.Error ||
-                     whereApparentOption == NotificationOption.Warning || whereApparentOption == NotificationOption.Error ||
-                     wherePossibleOption == NotificationOption.Warning || wherePossibleOption == NotificationOption.Error);
+            return !(forIntrinsicTypesOption == NotificationOption2.Warning || forIntrinsicTypesOption == NotificationOption2.Error ||
```

Comment:
💭 Can we used `ReportDiagnostic` instead of `NotificationOption2`?

Reply:
I'm fine with the naming used in this PR for now, especially considering the new types are internal. However, I would like to rename them in a follow-up PR (which will still be large, but much easier to review as a single change). We can discuss the proposed rename mapping later.

Reply:
Hmm - that is a good option but should be done in a follow-up PR as you mentioned to avoid semantic changes, which we want to minimize for this PR.